src/main/java/framework/pages/AbstractPage.java

```diff
@@ -0,0 +1,51 @@
+package framework.pages;
+
+import com.github.javafaker.Faker;
+import framework.components.Header;
+import framework.config.TestConfig;
+import framework.helpers.JavaScriptHelper;
+import framework.helpers.WebDriverWaitHelper;
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public abstract class AbstractPage {
+    protected WebDriver driver;
+    protected final String BASE_PAGE = TestConfig.CONFIG.baseUrl();
+    protected Faker faker;
+    protected JavaScriptHelper javaScriptHelper;
+
+    public AbstractPage(WebDriver driver) {
+        this.driver = driver;
+    }
```

Comment:
You can write

    protected AbstractPage() {
        PageFactory.initElements(WebDriverRunner.getDriver(), this);
        waitUntilLoaded();
    }

protected abstract void waitUntilLoaded();

in this case there is no need to add driver and init elements on each page

Reply:
Done. Thanks!